Lib/webbrowser.py

```diff
@@ -101,6 +123,12 @@ def open_new_tab(url):
     """
     return open(url, 2)
 
+def open_incognito(url):
+    """Open url in incognito mode of the default browser.
+
+    If not possible, then the behavior becomes equivalent to open_new().
```

Comment:
Falling back to "open_new" feels like "errors passing silently" to me. Why was this choice made?

Reply:
not all browsers support that. or some open a private tab by default (like opera) where it would make no sense. for browsers supporting such a feature it is implemented. so there is no "errors passing silently". please provide a use-case where it might fail if you disagree.

Reply:
As a user of the API I'd be disappointed if an URL is opened in regular mode when I explicitly asked to open it in incognito mode.  I'd therefore prefer to raise an exception when incognito mode isn't available.

For use cases where incognito mode is preferred but optional the user can code this themselves, whereas you cannot build my preferred semantics on top of the current implementation.

Reply:
well. when i look at the function doc and specification. the only thing i can say: "works as designed". 

but having `os.makedirs(..., exist_ok=True)` in mind; i would say: why not?!

you want it; you do it!

but keep it as default to raise no Exception (as it is just right now). since this also follows the new_tab, new_window approach to start a browser with a given url in any case. because this is (probably) mostlikely what comes to the closest what the user expects.

Reply:
The two API use cases are different: open_new_tab vs open_new is "just" a different look and feel. "open_incognito" has semantic differences in the browser state (such as not storing the page in long term history and separate cookie  stores).  Opening a page in a regular window when an incognito window was requested can result in unwanted side effects.

I wouldn't mind a keyword argument to enable a fallback to open_new, but it shouldn't be the default. Just like ``exists_ok`` defaults to false in ``os.makedirs``.  Something like ``def open_incognito(url, *, regular_ok=False)``.

Reply:
I agree with @ronaldoussoren: if a user is asking to keep their private URL secret, we must not "leak" that to the open internet, which could have extremely serious consequences for some people. Let's raise an exception instead.

Regarding Opera: if the user asks for a regular tab and it opens in private tab, that's fine, nothing is leaked.

Reply:
fine. 2 of 3 (@arhadthedev has not replied so far) voted. 

@hugovk could you please write the missing documentation so that we know how it works and how the function is defined. 

that also would fit into a "documentation first" development flow. 
as soon documentation is available @ronaldoussoren could start here to send his proposal.

i just saw that the issue to this PR was tagged with 3.13 - so - we spent already a lot of time talking. hopefully we not behind schedule now.

Reply:
We've plenty of time: 3.13 will be released in October 2024 with a May 2024 deadline to get new features in. 

For documentation, something like:

> `webbrowser.open_private(url) `
> Open *url* in a private/incognito window of the default browser, if possible, otherwise raise `RuntimeError`.

@ronaldoussoren has already made the proposal, @kr-g please could you update this PR?

Reply:
I have no preference on whether it should be an exception of a separate method or a permissive argument.

*I can agree with others about the issue in general only because a user who is supposed to get the incognito mode most likely will be unaware of it. So we have a choice between "always show user a warning in advance" vs "let the application gracefully treat a failure by showing an absolutely relevant warning with instructions (or an error if the very fact of page visitation must not be leaked)". Personally, I prefer the second option.*

The exact interface choice is out of my competence.